### PR TITLE
#110998 CW admin: add/edit tab

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: projects/valtimo/components/src/lib/components/breadcrumb-navigation/breadcrumb.service.ts [hunk 1]
```diff
@@ -29,30 +29,38 @@ import {TranslateService} from '@ngx-translate/core';
...
+        const secondBreadCrumb = this.getSecondBreadcrumb();
+
+        const breadcrumbs = [
+          ...(activeParentSequenceNumber ? [activeParentBreadcrumbItem] : []),
+          ...(manualSecondBreadcrumb ? [manualSecondBreadcrumb] : []),
+          ...(secondBreadCrumb && !manualSecondBreadcrumb ? [secondBreadCrumb] : []),
+        ];
+        if (manualThirdBreadcrumb) {
+          breadcrumbs.push(manualThirdBreadcrumb);
+        }
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/breadcrumb-navigation/breadcrumb.service.ts]:
```suggestion
          ...(activeParentSequenceNumber ? [activeParentBreadcrumbItem] : []),
          ...(manualSecondBreadcrumb ? [manualSecondBreadcrumb] : []),
          ...(secondBreadCrumb && !manualSecondBreadcrumb ? [secondBreadCrumb] : []),
          ...(!!manualThirdBreadcrumb ? [manualThirdBreadcrumb] : []),
        ];
```

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  </section>
+
+  <cds-modal-footer>
+    <ng-container *ngTemplateOutlet="editWidgetTabButtons;context:{obs: obs}"></ng-container>
+  </cds-modal-footer>
+</cds-modal>
+
+<ng-template #editWidgetTabFormTemplate let-obs="obs">
+  <form *ngIf="editWidgetTabForm" [formGroup]="editWidgetTabForm" class="widget-tab-modal__form">
+    <div class="form__element">
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.html]: This div should not be necessary, add this class directly to the cds-label element

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  &__row {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: flex-end;
+  }
+
+  &__element {
+    width: 100%;
+    margin-bottom: 32px;
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss]: If you want to distance the form from the footer, this should be done from the parent, the section element. Doing it from the element within the form exposes it to future alignment issues, should the template change

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  @Input() public widgetTab: WidgetTabItem;
+  @Output() public saveEvent = new EventEmitter<any>();
+
+  public readonly open$ = new BehaviorSubject<boolean>(false);
+  public readonly disabled$ = new BehaviorSubject<boolean>(false);
+  public readonly editWidgetTabForm = this.fb.group({
+    name: this.fb.control('', [Validators.required]),
+  });
+
+  public get widgetTabName() {
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts]: Return type

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  private setEditWidgetTabForm(): void {
+    if (this.widgetTab) {
+      this.widgetTabName?.setValue(this.widgetTab.name);
+    }
+
+    this.enable();
+  }
+
+  private openOpenSubscription(): void {
+    this._openSubscription = this.showModal$.subscribe(show => {
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts:104]: This is never unsubscribed, make sure to do it in OnDestroy

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<ng-container
+  *ngIf="{
+    currentWidgetTab: currentWidgetTab$ | async
+  } as obs"
+>
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.html]: If this is the only observable that's gonna go in here, then the obs object is not necessary. Please use currentWidgetTab$ as compactMode$ is used on line 34

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.scss
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.scss:1]: Remove empty file

File: projects/valtimo/components/src/lib/components/breadcrumb-navigation/breadcrumb.service.ts [hunk 7]
```diff
@@ -75,6 +83,15 @@ export class BreadcrumbService {
 
   clearSecondBreadcrumb(): void {
     this._manualSecondBreadcrumb$.next(null);
+    this._manualThirdBreadcrumb$.next(null);
```
Comment (mbritense) [inline projects/valtimo/components/src/lib/components/breadcrumb-navigation/breadcrumb.service.ts]: Change to clearThirdBreadcrumb(). Perhaps change clearSecondBreadcrumb to clearManualBreadcrumbs

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.edit-widget-tab-modal .form {
+  &__row {
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss:18]:
```suggestion
.edit-widget-tab-modal .form__row {
```

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: flex-end;
+  }
+}
```
Comment (teodor-ritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.scss]:
```suggestion
.edit-widget-tab-modal .form__row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;
}
```

**Review by teodor-ritense — APPROVED** [review]

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    private readonly fb: FormBuilder,
+    private readonly widgetTabManagementService: WidgetTabManagementService
+  ) {}
+
+  public ngOnInit(): void {
+    this.openOpenSubscription();
+  }
+
+  public ngOnDestroy(): void {
+    this._openSubscription.unsubscribe();
```
Comment (mbritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab-edit-modal/dossier-management-widget-tab-edit-modal.ts]: this._openSubscription?.unsubscribe();

File: projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  ~ https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" basis,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<ng-container>
```
Comment (mbritense) [inline projects/valtimo/dossier-management/src/lib/components/dossier-management-widget-tab/dossier-management-widget-tab.component.html]: remove empty ng-container

**Review by mbritense — APPROVED** [review]

**Pull request merged**